R/run.R

```diff
@@ -34,122 +30,120 @@
 #' name the SS3 file something besides `ss`, such as `ss3` or
 #' `ss_linux`.
 #'
-#' @author Ian G. Taylor, Kathryn L. Doering
+#' @author Ian G. Taylor, Kathryn L. Doering, Kelli F. Johnson
 #' @export
 #' @family run functions
 #' @examples
 #' \dontrun{
-#' dir <- file.path(system.file("extdata", package = "r4ss"), "simple_small")
+#' dir <- system.file("extdata/simple_small", package = "r4ss")
 #' r4ss::run(dir = dir)
 #' }
 #'
 run <- function(dir = getwd(),
                 exe = "ss",
                 extras = "",
                 skipfinished = TRUE,
-                show_in_console = TRUE,
+                show_in_console = FALSE,
                 console_output_file = "console.output.txt",
                 verbose = TRUE) {
   # check to make sure the first input is in the correct format
   if (!is.character(dir)) {
-    stop("Input 'dir' should be a character vector")
+    stop("Input 'dir' should be a character string")
   }
   if (!is.logical(show_in_console)) {
     stop("Input 'show_in_console' should be TRUE or FALSE")
   }
   if (!show_in_console & !is.character(console_output_file)) {
     stop("Input 'console_output_file' should be a character string")
   }
+  if (length(dir) > 1) {
+    lifecycle::deprecate_stop(
+      when = "1.46.2",
+      what = "run(dir = 'must be a single directory')",
+      details = "Please use the {purrr} package or other tools to apply run() to multiple directories"
+    )
+  }
 
   wd_orig <- getwd()
   on.exit(setwd(wd_orig), add = TRUE)
 
-  # vector of NA values to store results
-  results <- rep(NA, length(dir))
-
-  # this should always be "windows" or "unix" (includes Mac and Linux)
-  OS <- .Platform[["OS.type"]]
-
   # sort out path to executable
   check_exe_results <- check_exe(dir = dir, exe = exe, verbose = verbose)
   command <- check_exe_results[["exe"]]
 
-  # loop over directories
-  for (idir in seq_along(dir)) {
-    # directory where stuff will happen
-    dir <- dir[idir]
-
-    # confirm that dir exists
-    if (!dir.exists(dir[idir])) {
-      warning("not a directory:", dir[idir])
-      results[idir] <- "not a directory"
+  # confirm that dir exists
+  if (!dir.exists(dir)) {
+    warning("not a directory:", dir)
+    results <- "not a directory"
+  } else {
+    if (skipfinished & "Report.sso" %in% dir(dir)) {
```

Comment:
I think I might have mentioned this before but I believe this should be
```
if (file.exists(file.path(dir,"Report.sso")) && skipfinished) {
```
because you only care about skipfinished if the file actually exists. Furthermore, I think that using file.exists rather than %in% is more robust.